core/trino-main/src/main/java/io/trino/execution/SqlTaskManager.java

```diff
@@ -705,6 +706,29 @@ private Optional<StuckSplitTasksInterrupter> createStuckSplitTasksInterrupter(
                         taskExecutor));
     }
 
+    /**
+     * The class detect and interrupt runaway splits threads. It interrupts threads via failing the task that holding the split
+     * and relying on {@link PrioritizedSplitRunner#destroy()} method to actually interrupt the responsible thread.
+     * It detects at the frequency of {@link StuckSplitTasksInterrupter#stuckSplitsDetectionInterval}.
```

Comment:
Can we swap order of the statement with the next one? and minor reword suggestion of  something like:

```
The detection is invoked periodically with the frequency of {@link StuckSplitTasksInterrupter#stuckSplitsDetectionInterval}.
```